internal/file/file.go

```diff
@@ -13,9 +13,60 @@
 
 package file
 
-import "regexp"
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
 
 // IsValidFileName checks if a file name is cross-platform compatible
 func IsValidFileName(fileName string) bool {
 	return regexp.MustCompile(`^[a-zA-Z0-9_.-]+$`).MatchString(fileName)
 }
+
+// CopyToDir copies the src file to dst. Existing file will be overwritten.
+func CopyToDir(src, dst string) (int64, error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return 0, err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return 0, fmt.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return 0, err
+	}
+	defer source.Close()
+
+	if err := os.MkdirAll(dst, 0700); err != nil {
+		return 0, err
+	}
+	dstFile := filepath.Join(dst, filepath.Base(src))
```

Comment:
> Will this misinterpret file name `../../a.txt` ?

`filepath.Base(src)` gets the file name from `src`, which is `a.txt`. Then `filepath.Join` creates the dstFile with the same name but under directory `dst`, i.e., `<dst>/a.txt`.  So I don't think this will cause any issue?